fluent/ChainRune.ts

```diff
@@ -63,6 +63,11 @@ export class ChainRune<out C extends Chain, out U> extends Rune<C, U> {
     return Rune.rec({ connection, metadata }).into(this)
   }
 
+  with(connect: (signal: AbortSignal) => Connection) {
+    const connection = ConnectionRune.from(connect)
+    return Rune.rec({ connection, metadata: this.metadata }).into(ChainRune)
```

Comment:
To preserve the chain name, I believe you'll need
```suggestion
    return Rune.rec({ connection, metadata: this.metadata } as C).into(ChainRune)
```